### PR TITLE
chore: upgrade GitHub actions to solve deprecation warnings

### PR DESCRIPTION
**Pull request by algreasley**
Node updates necessitated GH Actions major upgrades .. no logic/config needed to change

### CODE REVIEW

File: packages/client/README.md [hunk 1]
```diff
@@ -26,7 +26,21 @@ The trading client GUI is a single page app (SPA) built using Typescript, React,
 
 Required:
 
-- [Node](https://nodejs.org) (v20+ see "engine" spec in package.json - anything between 16 and 20 is a disaster for OpenFin, due to problems with networking (DNS lookups) - suggest using `nvm`` to manage node instances)
+- [Node](https://nodejs.org) (v20+ see "engine" spec in package.json - suggest using `nvm` to manage node instances)
+
+VS Code / Plugins
+
+Suggest ESLint, Prettier extensions for immediate code quality management, with configuration such as:
```
Comment (algreasley) [inline packages/client/README.md:33]: just a handy hint I have been holding on to for a little while

File: packages/client/package-lock.json
Comment (algreasley) [inline packages/client/package-lock.json:1]: including a couple of minor updates to keep `npm audit` a bit happier

File: packages/client/package.json [hunk 2]
```diff
@@ -35,7 +35,7 @@
     "format": "npm run _format -- --write",
     "format:check": "npm run _format -- --check",
     "test": "vitest",
-    "verify": "npm run typecheck && npm run lint && npm run format:check && npm run test",
+    "verify": "npm run typecheck && npm run lint && npm run format:check && npm run test -- run",
```
Comment (algreasley) [inline packages/client/package.json:38]: the verify should always complete .. on CI it does already, ofc, but locally too

File: packages/client/package.json [hunk 3]
```diff
@@ -83,17 +83,17 @@
     "openfin-notifications": "1.23.0",
     "polished": "^4.1.1",
     "query-string": "^7.1.3",
-    "react": "18.2.0",
-    "react-dom": "18.2.0",
+    "react": "18.3.1",
```
Comment (algreasley) [inline packages/client/package.json:86]: bumping React to latest while we are here ...

File: packages/client/package.json [hunk 4]
```diff
@@ -83,17 +83,17 @@
...
+    "react-measure": "2.5.2",
     "react-router": "6.10.0",
     "react-router-dom": "6.10.0",
     "react-switch": "7.0.0",
     "react-virtualized-auto-sizer": "^1.0.7",
-    "react-window": "^1.8.8",
+    "react-window": "1.8.8",
     "ress": "^3.0.0",
-    "rxjs": "^7.5.4",
+    "rxjs": "7.8.1",
```
Comment (algreasley) [inline packages/client/package.json:96]: already at this version (latest v7), just making explicit

File: packages/client/src/client/styleguide/components/Input.tsx [hunk 5]
```diff
@@ -113,7 +113,7 @@ const Label = styled.label<LabelProps>`
 `
 
 export const Input = (
-  props: InputStyleProps & HTMLAttributes<HTMLInputElement>,
+  props: InputStyleProps & InputHTMLAttributes<HTMLInputElement>,
```
Comment (algreasley) [inline packages/client/src/client/styleguide/components/Input.tsx:116]: latest React types demand a more specific type here for inputs

File: packages/client/package.json [hunk 6]
```diff
@@ -123,16 +123,16 @@
...
     "cross-env": "^6.0.3",
     "csstype": "^3.0.2",
     "eslint": "8.38.0",
     "eslint-config-prettier": "8.8.0",
     "eslint-plugin-react": "7.32.2",
     "eslint-plugin-simple-import-sort": "^10.0.0",
     "jest-styled-components": "7.2.0",
-    "jsdom": "22.1.0",
-    "openfin-cli": "^4.0.0",
+    "jsdom": "24.1.0",
```
Comment (algreasley) [inline packages/client/package.json:134]: updated to remove a deprecation or two

File: packages/client/README.md [hunk 7]
```diff
@@ -26,7 +26,21 @@ The trading client GUI is a single page app (SPA) built using Typescript, React,
...
+
+```
+  "editor.codeActionsOnSave": {
+    "source.fixAll": "always",
+    "source.addMissingImports": "explicit"
+  },
+  "editor.formatOnSave": true,
+```
```
Comment (Dev-Alex23) [inline packages/client/README.md:41]: could add `json` to add some syntax highlighting. 
not necessary though.

File: packages/client/package.json [hunk 8]
```diff
@@ -83,17 +83,17 @@
     "openfin-notifications": "1.23.0",
     "polished": "^4.1.1",
     "query-string": "^7.1.3",
-    "react": "18.2.0",
-    "react-dom": "18.2.0",
+    "react": "18.3.1",
+    "react-dom": "18.3.1",
```
Comment (Dev-Alex23) [inline packages/client/package.json:87]: 19 should be out sometime this year.
Interested to see what's new and shiny

**Review by Dev-Alex23 — APPROVED** [review]
:raised_hands:

File: packages/client/package.json (hunk 8 above)
Comment (algreasley) [reply packages/client/package.json:87]: yep, RT will be there as soon as a stable v19 comes out .. they were explicitly advising against bumping to 19 for the pre-releases unless you are a library owner or similar

File: packages/client/README.md (hunk 7 above)
Comment (algreasley) [reply packages/client/README.md:41]: yep, will add soon .. and check other parts of the markdown too

**Pull request merged**